velox/exec/Window.cpp

```diff
@@ -340,8 +526,25 @@ void Window::callApplyForPartitionRows(
     auto rawFrameEndBuffer = frameEndBuffers_[w]->asMutable<vector_size_t>();
     rawFrameStartBuffers.push_back(rawFrameStartBuffer);
     rawFrameEndBuffers.push_back(rawFrameEndBuffer);
-  }
 
+    if (windowFrames_[w].startFrameArg) {
+      updateFrameVector(windowFrames_[w].startFrameArg.value());
+    }
+    if (windowFrames_[w].endFrameArg) {
+      updateFrameVector(windowFrames_[w].endFrameArg.value());
+    }
+  }
+  partitionOffset_ += numRows;
+
+  auto emptyFrameCheck = [&](vector_size_t& frameStart,
```

Comment:
With new design this check can be done once for the entire frame vector in one go.

Reply:
Thanks for the suggestion, will modify it once the new design is finalized.